src/main/java/subway/LineService.java

```diff
@@ -0,0 +1,94 @@
+package subway;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Line line = lineRepository.save(toEntity(lineRequest));
+        return createLineResponse(line);
+    }
+
+    private LineResponse createLineResponse(Line line) {
+        Station upStation = stationRepository.findById(line.getUpStationId())
+                .orElseThrow(() -> new IllegalArgumentException("해당 지하철 역은 존재하지 않습니다. id=" + line.getUpStationId()));
+
+        Station downStation = stationRepository.findById(line.getDownStationId())
+                .orElseThrow(() -> new IllegalArgumentException("해당 지하철 역은 존재하지 않습니다. id=" + line.getDownStationId()));
+
+        return new LineResponse(line.getId(), line.getName(), line.getColor(),
+                StationResponse.fromEntity(upStation), StationResponse.fromEntity(downStation), line.getDistance());
+    }
+
+    private Line toEntity(LineRequest lineRequest) {
+        return new Line(
+                lineRequest.getName(),
+                lineRequest.getColor(),
+                lineRequest.getUpStationId(),
+                lineRequest.getDownStationId(),
+                lineRequest.getDistance());
+    }
+
+    public List<LineResponse> findAllLines() {
+        List<Line> lines = lineRepository.findAll();
+        Map<Long, Station> stations = fetchStations(fetchStationsIds(lines));
+        return lines.stream().map(line ->
+                        new LineResponse(line.getId(), line.getName(), line.getColor(), line.getDistance(),
+                                createStationsResponses(stations, line))
+                )
+                .collect(Collectors.toList());
```

Comment:
```suggestion
        return lines.stream()
                .map(Line::getStationsIds)
                .flatMap(List::stream)
                .collect(Collectors.toList());
```

기교이지만, 이렇게도 가능해보이네요~

Reply:
사실 stream 으로 할려다가 자바 실력이 부족해서.. 수정했었습니다..
정말 정말 감사합니다!! 👍